### PR TITLE
[Improve] after change team the routers need to be regenerated

### LINKED ISSUE #1867: [Improve] after change team the router need to be regenerated
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

When switching teams, the backend to re-return the user's latest role, permissions, and router, the front-end needs to re-render the router

Currently, the backend does not return the latest router, The front-end also did not rebuild the generated router

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wolfboys**
When switching teams, the backend to re-return the user's latest role, permissions, and router, the front-end needs to re-render the router

Currently, the backend does not return the latest router, The front-end also did not rebuild the generated router

## What changes were proposed in this pull request

Issue Number: close #1867 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

File: streampark-console/streampark-console-webapp/src/store/modules/user.js [hunk 1]
```diff
@@ -61,6 +61,10 @@ const user = {
       storage.set(USER_ROUTER, routers)
       state.routers = routers
```
Comment (1996fanrui) [inline streampark-console/streampark-console-webapp/src/store/modules/user.js:72]: Should we use sessionStorage here? Mutil page shouldn't share the same router, right?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -203,6 +210,39 @@ public List<User> findByAppOwner(Long teamId) {
...
+        user.dataMasking();
+        userInfo.put("user", user);
+
+        // 3) roles
+        Set<String> roles = this.roleService.getUserRoleName(username);
+        userInfo.put("roles", roles);
+
+        // 4) permissions
+        Set<String> permissions = this.getPermissions(username);
+        userInfo.put("permissions", permissions);
```
Comment (1996fanrui) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java:241]: Could we return routers here? And remove getRouter.

As I understand, user, roles, permissions and routers should be return together after team is changed.

Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java:241]: > Could we return routers here? And remove getRouter.
> 
> As I understand, user, roles, permissions and routers should be return together after team is changed.

The latest routers need to request another interface, It is not returned with these bindings

File: streampark-console/streampark-console-webapp/src/store/modules/user.js (hunk 1 above)
Comment (wolfboys) [reply streampark-console/streampark-console-webapp/src/store/modules/user.js:72]: > Should we use sessionStorage here? Mutil page shouldn't share the same router, right?

oh, yes. you are right.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java (hunk 2 above)
Comment (1996fanrui) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java:241]: Why the routers cannot be return here? could you help share more information?

Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/UserServiceImpl.java:241]: > Why the routers cannot be return here? could you help share more information?

you can see: `getUserRouters` method in `MenuController` for more information.

**Comment by wolfboys** [comment]
hi @1996fanrui  all done. PTAL again

**Review by 1996fanrui — APPROVED** [review]
@wolfboys thanks for your contribution.

LGTM, however, we should refactor the front-end and back-end about getRouters in the future.

Current code is wired. The routers should be returned to front-end in the `setTeam` interface together.

**Review by GuoNingNing — APPROVED** [review]
LGTM

**Pull request merged**